src/main/java/com/google/edith/servlets/UserStatsServlet.java

```diff
@@ -26,21 +29,23 @@
 
   private final DatastoreService datastore;
   private final UserInsightsInterface userInsights;
+  private final String userId;
 
   public UserStatsServlet() {
     this.datastore = DatastoreServiceFactory.getDatastoreService();
     this.userInsights = new UserInsightsService();
+    this.userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();   
```

Comment:
This might be a bit less brittle if the UserService is a class variable/ dependency instead. This would give the caller control over what dependency to pass and make testing easier

Reply:
You can still set the userId as a class variable from the UserService var. Lmk if I missed anything